# Get BUSCO scores for all assemblies

* Benchmarking Universal Single-Copy Orthologs (BUSCO) from OrthoDB (www.orthodb.org) provides quantitative measures of the
completeness of transcriptome assemblies
* Eukaryota database used with 429 genes
* Metazoa database used with 843 genes
* ‘Complete’ lengths are within two standard deviations of the BUSCO group mean length
* Website: http://busco.ezlab.org/
* Simão et al. 2015: http://bioinformatics.oxfordjournals.org/content/31/19/3210
* Supplemental materials: http://busco.ezlab.org/files/BUSCO-SOM.pdf

On Jetstream (XSEDE, NSF), comamnd

```
busco -m trans -in /home/ljcohen/Porites_astreoides/Porites_astreoides_salinity.trinity.fixed.fasta \
-l ~/busco/metazoa -o P_ast_busco
```
Output results:

```
ljcohen@js-104-147:~/Porites_astreoides$ cat run*/short*
#Summarized BUSCO benchmarking for file: /home/ljcohen/Porites_astreoides/Porites_astreoides_salinity.trinity.fixed.fasta
#BUSCO was run in mode: trans

Summarized benchmarks in BUSCO notation:
        C:0%[D:0%],F:0%,M:0%,n:843

        152     Complete BUSCOs
        136     Complete and single-copy BUSCOs
        16      Complete and duplicated BUSCOs
        255     Fragmented BUSCOs
        436     Missing BUSCOs
        843     Total BUSCO groups searched
```


```
busco -m trans -in Porites_astreoides_salinity.trinity.fixed.fasta \
-l ~/busco/eukaryota -o P_ast_salinity_eukaryota

ljcohen@js-104-147:~/Porites_astreoides$ cat run_P_ast_salinity_eukaryota/short*
#Summarized BUSCO benchmarking for file: Porites_astreoides_salinity.trinity.fixed.fasta
#BUSCO was run in mode: trans

Summarized benchmarks in BUSCO notation:
        C:0%[D:0%],F:0%,M:0%,n:429

        101     Complete BUSCOs
        84      Complete and single-copy BUSCOs
        17      Complete and duplicated BUSCOs
        178     Fragmented BUSCOs
        150     Missing BUSCOs
        429     Total BUSCO groups searched

busco -m trans -in /home/ljcohen/Porites_astreoides/other_assemblies/mansour/Por_ast_ann_allTrans.fasta \
-l ~/busco/metazoa -o mansour_P_ast_busco_metaz

Summarized benchmarks in BUSCO notation:
        C:0%[D:0%],F:0%,M:0%,n:843

        786     Complete BUSCOs
        425     Complete and single-copy BUSCOs
        361     Complete and duplicated BUSCOs
        33      Fragmented BUSCOs
        24      Missing BUSCOs
        843     Total BUSCO groups searched

busco -m trans -in /home/ljcohen/Porites_astreoides/other_assemblies/mansour/Por_ast_ann_allTrans.fasta \
-l ~/busco/eukaryota -o mansour_P_ast_busco_euk

ljcohen@js-104-147:~/Porites_astreoides$ cat run_mansour_P_ast_busco_euk/short*
#Summarized BUSCO benchmarking for file: /home/ljcohen/Porites_astreoides/other_assemblies/mansour/Por_ast_ann_allTrans.fasta
#BUSCO was run in mode: trans

Summarized benchmarks in BUSCO notation:
        C:0%[D:0%],F:0%,M:0%,n:429

        395     Complete BUSCOs
        136     Complete and single-copy BUSCOs
        259     Complete and duplicated BUSCOs
        15      Fragmented BUSCOs
        19      Missing BUSCOs
        429     Total BUSCO groups searched


# Can use dammit to annotate

dammit annotate Porites_astreoides_larvae.Trinity.fasta --busco-group metazoa --database-dir /home/ljcohen/reference/dammit --n_threads 8


sed 's_|___g'
# diginorm reads, didn't finish got hung up so quit

transrate --assembly=Porites_astreoides_larvae.Trinity.fixed.fasta --threads=8 --left=/mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trinity/Porites_astreoides.left.fq --right=/mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trinity/Porites_astreoides.right.fq

# trimmed reads
transrate --assembly=Porites_astreoides_larvae.Trinity.fixed.fasta --threads=8 --left=/mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trim/all_1P.fq --right=/mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trim/all_2P.fq --output trim.P_ast.score

busco -m trans -in P_ast.SOAPdenovoTrans.NYUMC.fasta \
-l ~/busco/eukaryota -o NYUMC_SOAPdenovoTrans_P_ast_busco_euk

ljcohen@js-104-147:~/Porites_astreoides/other_assemblies/NYUMC$ cat run*/short*
#Summarized BUSCO benchmarking for file: P_ast.SOAPdenovoTrans.NYUMC.fasta
#BUSCO was run in mode: trans

Summarized benchmarks in BUSCO notation:
        C:0%[D:0%],F:0%,M:0%,n:429

        123     Complete BUSCOs
        109     Complete and single-copy BUSCOs
        14      Complete and duplicated BUSCOs
        124     Fragmented BUSCOs
        182     Missing BUSCOs
        429     Total BUSCO groups searched



busco -m trans -in P_ast.SOAPdenovoTrans.NYUMC.fasta \
-l ~/busco/metazoa -o NYUMC_SOAPdenovoTrans_P_ast_busco_metazoa

ljcohen@js-104-147:~/Porites_astreoides/other_assemblies/NYUMC$ cat run_NYUMC_*metazoa/short*
#Summarized BUSCO benchmarking for file: P_ast.SOAPdenovoTrans.NYUMC.fasta
#BUSCO was run in mode: trans

Summarized benchmarks in BUSCO notation:
        C:0%[D:0%],F:0%,M:0%,n:843

        183     Complete BUSCOs
        171     Complete and single-copy BUSCOs
        12      Complete and duplicated BUSCOs
        194     Fragmented BUSCOs
        466     Missing BUSCOs
        843     Total BUSCO groups searched

busco -m trans -in P_ast.Trinity.NYUMC.fasta \
-l ~/busco/metazoa -o NYUMC_Trinity_P_ast_busco_metazoa


ljcohen@js-104-147:~/Porites_astreoides/other_assemblies/NYUMC$ cat run_NYUMC_Trinity_P_ast_busco_metazoa/short*
#Summarized BUSCO benchmarking for file: P_ast.Trinity.NYUMC.fasta
#BUSCO was run in mode: trans

Summarized benchmarks in BUSCO notation:
        C:0%[D:0%],F:0%,M:0%,n:843

        305     Complete BUSCOs
        274     Complete and single-copy BUSCOs
        31      Complete and duplicated BUSCOs
        224     Fragmented BUSCOs
        314     Missing BUSCOs
        843     Total BUSCO groups searched

busco -m trans -in P_ast.Trinity.NYUMC.fasta \
-l ~/busco/eukaryota -o NYUMC_Trinity_P_ast_busco_eukaryota


ljcohen@js-104-147:~/Porites_astreoides/other_assemblies/NYUMC$ cat run_NYUMC_Trinity_P_ast_busco_eukaryota/short*
#Summarized BUSCO benchmarking for file: P_ast.Trinity.NYUMC.fasta
#BUSCO was run in mode: trans

Summarized benchmarks in BUSCO notation:
        C:0%[D:0%],F:0%,M:0%,n:429

        185     Complete BUSCOs
        161     Complete and single-copy BUSCOs
        24      Complete and duplicated BUSCOs
        125     Fragmented BUSCOs
        119     Missing BUSCOs
        429     Total BUSCO groups searched

# matz's assemblies

busco -m trans -in past.fasta \
-l ~/busco/eukaryota -o matz_past_eukaryota

ljcohen@js-104-147:~/Porites_astreoides/other_assemblies/matz$ cat run*/short*
#Summarized BUSCO benchmarking for file: past.fasta
#BUSCO was run in mode: trans

Summarized benchmarks in BUSCO notation:
        C:0%[D:0%],F:0%,M:0%,n:429

        71      Complete BUSCOs
        64      Complete and single-copy BUSCOs
        7       Complete and duplicated BUSCOs
        132     Fragmented BUSCOs
        226     Missing BUSCOs
        429     Total BUSCO groups searched

busco -m trans -in past.fasta \
> -l ~/busco/metazoa -o matz_past_metazoa


ljcohen@js-104-147:~/Porites_astreoides/other_assemblies/matz$ cat run_matz_past_metazoa/short*
#Summarized BUSCO benchmarking for file: past.fasta
#BUSCO was run in mode: trans

Summarized benchmarks in BUSCO notation:
        C:0%[D:0%],F:0%,M:0%,n:843

        87      Complete BUSCOs
        83      Complete and single-copy BUSCOs
        4       Complete and duplicated BUSCOs
        167     Fragmented BUSCOs
        589     Missing BUSCOs
        843     Total BUSCO groups searched


### This is accidentally from when I included --left and --right as the same read 1P:
# trimmed reads
transrate --assembly=Porites_astreoides_larvae.Trinity.fixed.fasta --threads=8 --left=/mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trim/all_1P.fq --right=/mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trim/all_1P.fq --output trim.P_ast.score

[ INFO] 2016-08-27 17:26:03 : Calculating contig metrics...
[ INFO] 2016-08-27 17:26:11 : Contig metrics:
[ INFO] 2016-08-27 17:26:11 : -----------------------------------
[ INFO] 2016-08-27 17:26:11 : n seqs                        61184
[ INFO] 2016-08-27 17:26:11 : smallest                        201
[ INFO] 2016-08-27 17:26:11 : largest                        6847
[ INFO] 2016-08-27 17:26:11 : n bases                    24600994
[ INFO] 2016-08-27 17:26:11 : mean len                     402.08
[ INFO] 2016-08-27 17:26:11 : n under 200                       0
[ INFO] 2016-08-27 17:26:11 : n over 1k                      2464
[ INFO] 2016-08-27 17:26:11 : n over 10k                        0
[ INFO] 2016-08-27 17:26:11 : n with orf                     7800
[ INFO] 2016-08-27 17:26:11 : mean orf percent              69.35
[ INFO] 2016-08-27 17:26:11 : n90                             229
[ INFO] 2016-08-27 17:26:11 : n70                             304
[ INFO] 2016-08-27 17:26:11 : n50                             422
[ INFO] 2016-08-27 17:26:11 : n30                             637
[ INFO] 2016-08-27 17:26:11 : n10                            1203
[ INFO] 2016-08-27 17:26:11 : gc                             0.42
[ INFO] 2016-08-27 17:26:11 : gc skew                        0.02
[ INFO] 2016-08-27 17:26:11 : at skew                        0.02
[ INFO] 2016-08-27 17:26:11 : cpg ratio                      1.71
[ INFO] 2016-08-27 17:26:11 : bases n                           0
[ INFO] 2016-08-27 17:26:11 : proportion n                    0.0
[ INFO] 2016-08-27 17:26:11 : linguistic complexity          0.09
[ INFO] 2016-08-27 17:26:11 : Contig metrics done in 9 seconds
[ INFO] 2016-08-27 17:26:11 : Calculating read diagnostics...
[ INFO] 2016-08-27 17:45:29 : Read mapping metrics:
[ INFO] 2016-08-27 17:45:29 : -----------------------------------
[ INFO] 2016-08-27 17:45:29 : fragments                  88774048
[ INFO] 2016-08-27 17:45:29 : fragments mapped           58974118
[ INFO] 2016-08-27 17:45:29 : p fragments mapped             0.66
[ INFO] 2016-08-27 17:45:29 : good mappings                 19771
[ INFO] 2016-08-27 17:45:29 : p good mapping                  0.0
[ INFO] 2016-08-27 17:45:29 : bad mappings               58954347
[ INFO] 2016-08-27 17:45:29 : potential bridges               610
[ INFO] 2016-08-27 17:45:29 : bases uncovered             2757502
[ INFO] 2016-08-27 17:45:29 : p bases uncovered              0.11
[ INFO] 2016-08-27 17:45:29 : contigs uncovbase             46170
[ INFO] 2016-08-27 17:45:29 : p contigs uncovbase            0.75
[ INFO] 2016-08-27 17:45:29 : contigs uncovered              7638
[ INFO] 2016-08-27 17:45:29 : p contigs uncovered            0.12
[ INFO] 2016-08-27 17:45:29 : contigs lowcovered             8949
[ INFO] 2016-08-27 17:45:29 : p contigs lowcovered           0.15
[ INFO] 2016-08-27 17:45:29 : contigs segmented              6213
[ INFO] 2016-08-27 17:45:29 : p contigs segmented             0.1
[ INFO] 2016-08-27 17:45:29 : Read metrics done in 1158 seconds
[ INFO] 2016-08-27 17:45:29 : No reference provided, skipping comparative diagnostics
[ INFO] 2016-08-27 17:45:30 : TRANSRATE ASSEMBLY SCORE        0.0
[ INFO] 2016-08-27 17:45:30 : -----------------------------------
[ INFO] 2016-08-27 17:45:30 : TRANSRATE OPTIMAL SCORE         0.0
[ INFO] 2016-08-27 17:45:30 : TRANSRATE OPTIMAL CUTOFF     0.2443
[ INFO] 2016-08-27 17:45:30 : good contigs                      9
[ INFO] 2016-08-27 17:45:30 : p good contigs                  0.0


#STAR (didn't work out)

STAR --runThreadN 8 --runMode genomeGenerate \
      --genomeDir /mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/reference/mansour/star_index --genomeFastaFiles /mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/reference/mansour/Por_ast_ann_allTrans.fasta

ran out of memory

STAR --runThreadN 8 --genomeDir /mnt/research/ged/lisa/hboi/NGS/sponge_OD_exposure/ref/sponge_star_index \
    --readFilesIn  /mnt/research/ged/lisa/hboi/NGS/sponge_OD_exposure/baseclear/raw_sequences/OC_GCCAAT_L002_R1_001_BD20BUACXX.filt.fastq /mnt/research/ged/lisa/hboi/NGS/sponge_OD_exposure/baseclear/raw_sequences/OC_GCCAAT_L002_R2_001_BD20BUACXX.filt.fastq \
    --outFileNamePrefix /mnt/research/ged/lisa/hboi/NGS/sponge_OD_exposure/sponge_OC --outSAMtype BAM Unsorted

bwa index Por_ast_ann_allTrans.fasta
bwa mem /mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/reference/mansour/Por_ast_ann_allTrans.fasta /mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trim/all_1P.fq /mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trim/all_2P.fq > P_ast_salinity.bam 

[ljcohen@dev-intel14 2016]$ cat P_ast_flagstat.txt
86087723 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 secondary
1756 + 0 supplementary
0 + 0 duplicates
84187910 + 0 mapped (97.79%:-nan%)
86085967 + 0 paired in sequencing
43042984 + 0 read1
43042983 + 0 read2
81220241 + 0 properly paired (94.35%:-nan%)
83608687 + 0 with itself and mate mapped
577467 + 0 singletons (0.67%:-nan%)
1596684 + 0 with mate mapped to a different chr
663241 + 0 with mate mapped to a different chr (mapQ>=5)


# what about transrate reference, reciprocal?
mapping is almost 100% with Tamer's reference using bwa (what about bowtie?)
Do we have anything new?

transrate --assembly=Porites_astreoides_larvae.Trinity.fixed.fasta --threads=8 --reference /mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/reference/mansour/Por_ast_ann_allTrans.fasta --output salinity_v_mansour
[ INFO] 2016-08-27 20:36:21 : n seqs                        61184
[ INFO] 2016-08-27 20:36:21 : smallest                        201
[ INFO] 2016-08-27 20:36:21 : largest                        6847
[ INFO] 2016-08-27 20:36:21 : n bases                    24600994
[ INFO] 2016-08-27 20:36:21 : mean len                     402.08
[ INFO] 2016-08-27 20:36:21 : n under 200                       0
[ INFO] 2016-08-27 20:36:21 : n over 1k                      2464
[ INFO] 2016-08-27 20:36:21 : n over 10k                        0
[ INFO] 2016-08-27 20:36:21 : n with orf                     7800
[ INFO] 2016-08-27 20:36:21 : mean orf percent              69.35
[ INFO] 2016-08-27 20:36:21 : n90                             229
[ INFO] 2016-08-27 20:36:21 : n70                             304
[ INFO] 2016-08-27 20:36:21 : n50                             422
[ INFO] 2016-08-27 20:36:21 : n30                             637
[ INFO] 2016-08-27 20:36:21 : n10                            1203
[ INFO] 2016-08-27 20:36:21 : gc                             0.42
[ INFO] 2016-08-27 20:36:21 : gc skew                        0.02
[ INFO] 2016-08-27 20:36:21 : at skew                        0.02
[ INFO] 2016-08-27 20:36:21 : cpg ratio                      1.71
[ INFO] 2016-08-27 20:36:21 : bases n                           0
[ INFO] 2016-08-27 20:36:21 : proportion n                    0.0
[ INFO] 2016-08-27 20:36:21 : linguistic complexity          0.09
[ INFO] 2016-08-27 20:36:21 : Contig metrics done in 13 seconds
[ INFO] 2016-08-27 20:36:21 : No reads provided, skipping read diagnostics
[ INFO] 2016-08-27 20:36:21 : Calculating comparative metrics...
[ INFO] 2016-08-27 20:45:14 : Comparative metrics:
[ INFO] 2016-08-27 20:45:14 : -----------------------------------
[ INFO] 2016-08-27 20:45:14 : CRBB hits                     54969
[ INFO] 2016-08-27 20:45:14 : n contigs with CRBB           54969
[ INFO] 2016-08-27 20:45:14 : p contigs with CRBB             0.9
[ INFO] 2016-08-27 20:45:14 : rbh per reference              0.06
[ INFO] 2016-08-27 20:45:14 : n refs with CRBB              25718
[ INFO] 2016-08-27 20:45:14 : p refs with CRBB               0.03
[ INFO] 2016-08-27 20:45:14 : cov25                          8304
[ INFO] 2016-08-27 20:45:14 : p cov25                        0.01
[ INFO] 2016-08-27 20:45:14 : cov50                          3524
[ INFO] 2016-08-27 20:45:14 : p cov50                         0.0
[ INFO] 2016-08-27 20:45:14 : cov75                          1253
[ INFO] 2016-08-27 20:45:14 : p cov75                         0.0
[ INFO] 2016-08-27 20:45:14 : cov85                           709
[ INFO] 2016-08-27 20:45:14 : p cov85                         0.0
[ INFO] 2016-08-27 20:45:14 : cov95                           316
[ INFO] 2016-08-27 20:45:14 : p cov95                         0.0
[ INFO] 2016-08-27 20:45:14 : reference coverage             0.02
[ INFO] 2016-08-27 20:45:14 : Comparative metrics done in 533 seconds

# reverse
transrate --assembly /mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/reference/mansour/Por_ast_ann_allTrans.fasta --threads=8 --reference Porites_astreoides_larvae.Trinity.fixed.fasta --output mansour_v_salinity

[ INFO] 2016-08-27 20:55:35 : Contig metrics:
[ INFO] 2016-08-27 20:55:35 : -----------------------------------
[ INFO] 2016-08-27 20:55:35 : n seqs                       868830
[ INFO] 2016-08-27 20:55:35 : smallest                        201
[ INFO] 2016-08-27 20:55:35 : largest                       28297
[ INFO] 2016-08-27 20:55:35 : n bases                   595065082
[ INFO] 2016-08-27 20:55:35 : mean len                      684.9
[ INFO] 2016-08-27 20:55:35 : n under 200                       0
[ INFO] 2016-08-27 20:55:35 : n over 1k                    145908
[ INFO] 2016-08-27 20:55:35 : n over 10k                      319
[ INFO] 2016-08-27 20:55:35 : n with orf                   197152
[ INFO] 2016-08-27 20:55:35 : mean orf percent              61.34
[ INFO] 2016-08-27 20:55:35 : n90                             287
[ INFO] 2016-08-27 20:55:35 : n70                             513
[ INFO] 2016-08-27 20:55:35 : n50                            1004
[ INFO] 2016-08-27 20:55:35 : n30                            1870
[ INFO] 2016-08-27 20:55:35 : n10                            3874
[ INFO] 2016-08-27 20:55:35 : gc                             0.44
[ INFO] 2016-08-27 20:55:35 : gc skew                        0.01
[ INFO] 2016-08-27 20:55:35 : at skew                         0.0
[ INFO] 2016-08-27 20:55:35 : cpg ratio                      1.91
[ INFO] 2016-08-27 20:55:35 : bases n                           0
[ INFO] 2016-08-27 20:55:35 : proportion n                    0.0
[ INFO] 2016-08-27 20:55:35 : linguistic complexity          0.13
[ INFO] 2016-08-27 20:55:35 : Contig metrics done in 272 seconds
[ INFO] 2016-08-27 20:55:35 : No reads provided, skipping read diagnostics
[ INFO] 2016-08-27 20:55:35 : Calculating comparative metrics...
[ INFO] 2016-08-27 21:04:21 : Comparative metrics:
[ INFO] 2016-08-27 21:04:21 : -----------------------------------
[ INFO] 2016-08-27 21:04:21 : CRBB hits                     89268
[ INFO] 2016-08-27 21:04:21 : n contigs with CRBB           89268
[ INFO] 2016-08-27 21:04:21 : p contigs with CRBB             0.1
[ INFO] 2016-08-27 21:04:21 : rbh per reference              1.46
[ INFO] 2016-08-27 21:04:21 : n refs with CRBB              31808
[ INFO] 2016-08-27 21:04:21 : p refs with CRBB               0.52
[ INFO] 2016-08-27 21:04:21 : cov25                         13751
[ INFO] 2016-08-27 21:04:21 : p cov25                        0.22
[ INFO] 2016-08-27 21:04:21 : cov50                         11366
[ INFO] 2016-08-27 21:04:21 : p cov50                        0.19
[ INFO] 2016-08-27 21:04:21 : cov75                         10553
[ INFO] 2016-08-27 21:04:21 : p cov75                        0.17
[ INFO] 2016-08-27 21:04:21 : cov85                         10197
[ INFO] 2016-08-27 21:04:21 : p cov85                        0.17
[ INFO] 2016-08-27 21:04:21 : cov95                          9509
[ INFO] 2016-08-27 21:04:21 : p cov95                        0.16
[ INFO] 2016-08-27 21:04:21 : reference coverage             0.25
[ INFO] 2016-08-27 21:04:21 : Comparative metrics done in 525 seconds


transrate --assembly=Porites_astreoides_larvae.Trinity.fixed.fasta --threads=8 --reference /mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/reference/mansour/p_ast2016.fasta --output P_ast_salinity_v_mansour_Past_only

INFO] 2016-08-27 21:27:33 : Contig metrics:
[ INFO] 2016-08-27 21:27:33 : -----------------------------------
[ INFO] 2016-08-27 21:27:33 : n seqs                        61184
[ INFO] 2016-08-27 21:27:33 : smallest                        201
[ INFO] 2016-08-27 21:27:33 : largest                        6847
[ INFO] 2016-08-27 21:27:33 : n bases                    24600994
[ INFO] 2016-08-27 21:27:33 : mean len                     402.08
[ INFO] 2016-08-27 21:27:33 : n under 200                       0
[ INFO] 2016-08-27 21:27:33 : n over 1k                      2464
[ INFO] 2016-08-27 21:27:33 : n over 10k                        0
[ INFO] 2016-08-27 21:27:33 : n with orf                     7800
[ INFO] 2016-08-27 21:27:33 : mean orf percent              69.35
[ INFO] 2016-08-27 21:27:33 : n90                             229
[ INFO] 2016-08-27 21:27:33 : n70                             304
[ INFO] 2016-08-27 21:27:33 : n50                             422
[ INFO] 2016-08-27 21:27:33 : n30                             637
[ INFO] 2016-08-27 21:27:33 : n10                            1203
[ INFO] 2016-08-27 21:27:33 : gc                             0.42
[ INFO] 2016-08-27 21:27:33 : gc skew                        0.02
[ INFO] 2016-08-27 21:27:33 : at skew                        0.02
[ INFO] 2016-08-27 21:27:33 : cpg ratio                      1.71
[ INFO] 2016-08-27 21:27:33 : bases n                           0
[ INFO] 2016-08-27 21:27:33 : proportion n                    0.0
[ INFO] 2016-08-27 21:27:33 : linguistic complexity          0.09
[ INFO] 2016-08-27 21:27:33 : Contig metrics done in 10 seconds
[ INFO] 2016-08-27 21:27:33 : No reads provided, skipping read diagnostics
[ INFO] 2016-08-27 21:27:33 : Calculating comparative metrics...
[ INFO] 2016-08-27 21:29:06 : Comparative metrics:
[ INFO] 2016-08-27 21:29:06 : -----------------------------------
[ INFO] 2016-08-27 21:29:06 : CRBB hits                     25613
[ INFO] 2016-08-27 21:29:06 : n contigs with CRBB           25613
[ INFO] 2016-08-27 21:29:06 : p contigs with CRBB            0.42
[ INFO] 2016-08-27 21:29:06 : rbh per reference               0.2
[ INFO] 2016-08-27 21:29:06 : n refs with CRBB              15246
[ INFO] 2016-08-27 21:29:06 : p refs with CRBB               0.12
[ INFO] 2016-08-27 21:29:06 : cov25                          3109
[ INFO] 2016-08-27 21:29:06 : p cov25                        0.02
[ INFO] 2016-08-27 21:29:06 : cov50                          1251
[ INFO] 2016-08-27 21:29:06 : p cov50                        0.01
[ INFO] 2016-08-27 21:29:06 : cov75                           461
[ INFO] 2016-08-27 21:29:06 : p cov75                         0.0
[ INFO] 2016-08-27 21:29:06 : cov85                           249
[ INFO] 2016-08-27 21:29:06 : p cov85                         0.0
[ INFO] 2016-08-27 21:29:06 : cov95                           137
[ INFO] 2016-08-27 21:29:06 : p cov95                         0.0
[ INFO] 2016-08-27 21:29:06 : reference coverage             0.02
[ INFO] 2016-08-27 21:29:06 : Comparative metrics done in 93 seconds

transrate --assembly /mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/reference/mansour/p_ast2016.fasta --threads=8 --reference Porites_astreoides_larvae.Trinity.fixed.fasta --output mansour_Past_only_v_P_ast_salinity


[ INFO] 2016-08-27 21:30:31 : Contig metrics:
[ INFO] 2016-08-27 21:30:31 : -----------------------------------
[ INFO] 2016-08-27 21:30:31 : n seqs                       129718
[ INFO] 2016-08-27 21:30:31 : smallest                        201
[ INFO] 2016-08-27 21:30:31 : largest                       19877
[ INFO] 2016-08-27 21:30:31 : n bases                   105218865
[ INFO] 2016-08-27 21:30:31 : mean len                     811.14
[ INFO] 2016-08-27 21:30:31 : n under 200                       0
[ INFO] 2016-08-27 21:30:31 : n over 1k                     27334
[ INFO] 2016-08-27 21:30:31 : n over 10k                       58
[ INFO] 2016-08-27 21:30:31 : n with orf                    22182
[ INFO] 2016-08-27 21:30:31 : mean orf percent              46.48
[ INFO] 2016-08-27 21:30:31 : n90                             316
[ INFO] 2016-08-27 21:30:31 : n70                             654
[ INFO] 2016-08-27 21:30:31 : n50                            1358
[ INFO] 2016-08-27 21:30:31 : n30                            2487
[ INFO] 2016-08-27 21:30:31 : n10                            4642
[ INFO] 2016-08-27 21:30:31 : gc                              0.4
[ INFO] 2016-08-27 21:30:31 : gc skew                        0.01
[ INFO] 2016-08-27 21:30:31 : at skew                         0.0
[ INFO] 2016-08-27 21:30:31 : cpg ratio                      1.78
[ INFO] 2016-08-27 21:30:31 : bases n                           0
[ INFO] 2016-08-27 21:30:31 : proportion n                    0.0
[ INFO] 2016-08-27 21:30:31 : linguistic complexity          0.15
[ INFO] 2016-08-27 21:30:31 : Contig metrics done in 24 seconds
[ INFO] 2016-08-27 21:30:31 : No reads provided, skipping read diagnostics
[ INFO] 2016-08-27 21:30:31 : Calculating comparative metrics...
[ INFO] 2016-08-27 21:31:46 : Comparative metrics:
[ INFO] 2016-08-27 21:31:46 : -----------------------------------
[ INFO] 2016-08-27 21:31:46 : CRBB hits                     35914
[ INFO] 2016-08-27 21:31:46 : n contigs with CRBB           35914
[ INFO] 2016-08-27 21:31:46 : p contigs with CRBB            0.28
[ INFO] 2016-08-27 21:31:46 : rbh per reference              0.59
[ INFO] 2016-08-27 21:31:46 : n refs with CRBB              16142
[ INFO] 2016-08-27 21:31:46 : p refs with CRBB               0.26
[ INFO] 2016-08-27 21:31:46 : cov25                          5331
[ INFO] 2016-08-27 21:31:46 : p cov25                        0.09
[ INFO] 2016-08-27 21:31:46 : cov50                          3792
[ INFO] 2016-08-27 21:31:46 : p cov50                        0.06
[ INFO] 2016-08-27 21:31:46 : cov75                          3210
[ INFO] 2016-08-27 21:31:46 : p cov75                        0.05
[ INFO] 2016-08-27 21:31:46 : cov85                          2993
[ INFO] 2016-08-27 21:31:46 : p cov85                        0.05
[ INFO] 2016-08-27 21:31:46 : cov95                          2666
[ INFO] 2016-08-27 21:31:46 : p cov95                        0.04
[ INFO] 2016-08-27 21:31:46 : reference coverage             0.08
[ INFO] 2016-08-27 21:31:46 : Comparative metrics done in 76 seconds

transrate --assembly Porites_astreoides_larvae.Trinity.fixed.fasta --threads=8 --reference /mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/reference/mansour/trinityVsSymb_best.sig.fasta --output P_ast_salinity_v_mansour_Symbio


[ INFO] 2016-08-27 21:34:50 : Contig metrics:
[ INFO] 2016-08-27 21:34:50 : -----------------------------------
[ INFO] 2016-08-27 21:34:50 : n seqs                        61184
[ INFO] 2016-08-27 21:34:50 : smallest                        201
[ INFO] 2016-08-27 21:34:50 : largest                        6847
[ INFO] 2016-08-27 21:34:50 : n bases                    24600994
[ INFO] 2016-08-27 21:34:50 : mean len                     402.08
[ INFO] 2016-08-27 21:34:50 : n under 200                       0
[ INFO] 2016-08-27 21:34:50 : n over 1k                      2464
[ INFO] 2016-08-27 21:34:50 : n over 10k                        0
[ INFO] 2016-08-27 21:34:50 : n with orf                     7800
[ INFO] 2016-08-27 21:34:50 : mean orf percent              69.35
[ INFO] 2016-08-27 21:34:50 : n90                             229
[ INFO] 2016-08-27 21:34:50 : n70                             304
[ INFO] 2016-08-27 21:34:50 : n50                             422
[ INFO] 2016-08-27 21:34:50 : n30                             637
[ INFO] 2016-08-27 21:34:50 : n10                            1203
[ INFO] 2016-08-27 21:34:50 : gc                             0.42
[ INFO] 2016-08-27 21:34:50 : gc skew                        0.02
[ INFO] 2016-08-27 21:34:50 : at skew                        0.02
[ INFO] 2016-08-27 21:34:50 : cpg ratio                      1.71
[ INFO] 2016-08-27 21:34:50 : bases n                           0
[ INFO] 2016-08-27 21:34:50 : proportion n                    0.0
[ INFO] 2016-08-27 21:34:50 : linguistic complexity          0.09
[ INFO] 2016-08-27 21:34:50 : Contig metrics done in 8 seconds
[ INFO] 2016-08-27 21:34:50 : No reads provided, skipping read diagnostics
[ INFO] 2016-08-27 21:34:50 : Calculating comparative metrics...
[ INFO] 2016-08-27 21:37:54 : Comparative metrics:
[ INFO] 2016-08-27 21:37:54 : -----------------------------------
[ INFO] 2016-08-27 21:37:54 : CRBB hits                     44813
[ INFO] 2016-08-27 21:37:54 : n contigs with CRBB           44813
[ INFO] 2016-08-27 21:37:54 : p contigs with CRBB            0.73
[ INFO] 2016-08-27 21:37:54 : rbh per reference              0.24
[ INFO] 2016-08-27 21:37:54 : n refs with CRBB              18906
[ INFO] 2016-08-27 21:37:54 : p refs with CRBB                0.1
[ INFO] 2016-08-27 21:37:54 : cov25                          6445
[ INFO] 2016-08-27 21:37:54 : p cov25                        0.03
[ INFO] 2016-08-27 21:37:54 : cov50                          2783
[ INFO] 2016-08-27 21:37:54 : p cov50                        0.01
[ INFO] 2016-08-27 21:37:54 : cov75                           973
[ INFO] 2016-08-27 21:37:54 : p cov75                        0.01
[ INFO] 2016-08-27 21:37:54 : cov85                           552
[ INFO] 2016-08-27 21:37:54 : p cov85                         0.0
[ INFO] 2016-08-27 21:37:54 : cov95                           238
[ INFO] 2016-08-27 21:37:54 : p cov95                         0.0
[ INFO] 2016-08-27 21:37:54 : reference coverage             0.04
[ INFO] 2016-08-27 21:37:54 : Comparative metrics done in 184 seconds

transrate --assembly /mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/reference/mansour/trinityVsSymb_best.sig.fasta --threads=8 --reference Porites_astreoides_larvae.Trinity.fixed.fasta  --output mansour_Symbio_v_Past_salinity

[ INFO] 2016-08-27 21:39:47 : Contig metrics:
[ INFO] 2016-08-27 21:39:47 : -----------------------------------
[ INFO] 2016-08-27 21:39:47 : n seqs                       186177
[ INFO] 2016-08-27 21:39:47 : smallest                        201
[ INFO] 2016-08-27 21:39:47 : largest                       28297
[ INFO] 2016-08-27 21:39:47 : n bases                   205741294
[ INFO] 2016-08-27 21:39:47 : mean len                    1105.08
[ INFO] 2016-08-27 21:39:47 : n under 200                       0
[ INFO] 2016-08-27 21:39:47 : n over 1k                     67354
[ INFO] 2016-08-27 21:39:47 : n over 10k                      260
[ INFO] 2016-08-27 21:39:47 : n with orf                    72143
[ INFO] 2016-08-27 21:39:47 : mean orf percent              60.39
[ INFO] 2016-08-27 21:39:47 : n90                             440
[ INFO] 2016-08-27 21:39:47 : n70                            1097
[ INFO] 2016-08-27 21:39:47 : n50                            1804
[ INFO] 2016-08-27 21:39:47 : n30                            2825
[ INFO] 2016-08-27 21:39:47 : n10                            5230
[ INFO] 2016-08-27 21:39:47 : gc                             0.48
[ INFO] 2016-08-27 21:39:47 : gc skew                        -0.0
[ INFO] 2016-08-27 21:39:47 : at skew                         0.0
[ INFO] 2016-08-27 21:39:47 : cpg ratio                      1.95
[ INFO] 2016-08-27 21:39:47 : bases n                           0
[ INFO] 2016-08-27 21:39:47 : proportion n                    0.0
[ INFO] 2016-08-27 21:39:47 : linguistic complexity          0.19
[ INFO] 2016-08-27 21:39:47 : Contig metrics done in 38 seconds
[ INFO] 2016-08-27 21:39:47 : No reads provided, skipping read diagnostics
[ INFO] 2016-08-27 21:39:47 : Calculating comparative metrics...
[ INFO] 2016-08-27 21:42:24 : Comparative metrics:
[ INFO] 2016-08-27 21:42:24 : -----------------------------------
[ INFO] 2016-08-27 21:42:24 : CRBB hits                     50056
[ INFO] 2016-08-27 21:42:24 : n contigs with CRBB           50056
[ INFO] 2016-08-27 21:42:24 : p contigs with CRBB            0.27
[ INFO] 2016-08-27 21:42:24 : rbh per reference              0.82
[ INFO] 2016-08-27 21:42:24 : n refs with CRBB              20387
[ INFO] 2016-08-27 21:42:24 : p refs with CRBB               0.33
[ INFO] 2016-08-27 21:42:24 : cov25                          9092
[ INFO] 2016-08-27 21:42:24 : p cov25                        0.15
[ INFO] 2016-08-27 21:42:24 : cov50                          7955
[ INFO] 2016-08-27 21:42:24 : p cov50                        0.13
[ INFO] 2016-08-27 21:42:24 : cov75                          7368
[ INFO] 2016-08-27 21:42:24 : p cov75                        0.12
[ INFO] 2016-08-27 21:42:24 : cov85                          7107
[ INFO] 2016-08-27 21:42:24 : p cov85                        0.12
[ INFO] 2016-08-27 21:42:24 : cov95                          6643
[ INFO] 2016-08-27 21:42:24 : p cov95                        0.11
[ INFO] 2016-08-27 21:42:24 : reference coverage             0.18


# matz
transrate --assembly Porites_astreoides_larvae.Trinity.fixed.fasta --threads=8 --reference /mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/reference/matz/past.fasta --output P_ast_salinity_v_matz


27 21:46:47 : Contig metrics:
[ INFO] 2016-08-27 21:46:47 : -----------------------------------
[ INFO] 2016-08-27 21:46:47 : n seqs                        61184
[ INFO] 2016-08-27 21:46:47 : smallest                        201
[ INFO] 2016-08-27 21:46:47 : largest                        6847
[ INFO] 2016-08-27 21:46:47 : n bases                    24600994
[ INFO] 2016-08-27 21:46:47 : mean len                     402.08
[ INFO] 2016-08-27 21:46:47 : n under 200                       0
[ INFO] 2016-08-27 21:46:47 : n over 1k                      2464
[ INFO] 2016-08-27 21:46:47 : n over 10k                        0
[ INFO] 2016-08-27 21:46:47 : n with orf                     7800
[ INFO] 2016-08-27 21:46:47 : mean orf percent              69.35
[ INFO] 2016-08-27 21:46:47 : n90                             229
[ INFO] 2016-08-27 21:46:47 : n70                             304
[ INFO] 2016-08-27 21:46:47 : n50                             422
[ INFO] 2016-08-27 21:46:47 : n30                             637
[ INFO] 2016-08-27 21:46:47 : n10                            1203
[ INFO] 2016-08-27 21:46:47 : gc                             0.42
[ INFO] 2016-08-27 21:46:47 : gc skew                        0.02
[ INFO] 2016-08-27 21:46:47 : at skew                        0.02
[ INFO] 2016-08-27 21:46:47 : cpg ratio                      1.71
[ INFO] 2016-08-27 21:46:47 : bases n                           0
[ INFO] 2016-08-27 21:46:47 : proportion n                    0.0
[ INFO] 2016-08-27 21:46:47 : linguistic complexity          0.09
[ INFO] 2016-08-27 21:46:47 : Contig metrics done in 10 seconds
[ INFO] 2016-08-27 21:46:47 : No reads provided, skipping read diagnostics
[ INFO] 2016-08-27 21:46:47 : Calculating comparative metrics...
[ INFO] 2016-08-27 21:47:17 : Comparative metrics:
[ INFO] 2016-08-27 21:47:17 : -----------------------------------
[ INFO] 2016-08-27 21:47:17 : CRBB hits                     22975
[ INFO] 2016-08-27 21:47:17 : n contigs with CRBB           22975
[ INFO] 2016-08-27 21:47:17 : p contigs with CRBB            0.38
[ INFO] 2016-08-27 21:47:17 : rbh per reference              0.75
[ INFO] 2016-08-27 21:47:17 : n refs with CRBB              13618
[ INFO] 2016-08-27 21:47:17 : p refs with CRBB               0.44
[ INFO] 2016-08-27 21:47:17 : cov25                          6438
[ INFO] 2016-08-27 21:47:17 : p cov25                        0.21
[ INFO] 2016-08-27 21:47:17 : cov50                          4369
[ INFO] 2016-08-27 21:47:17 : p cov50                        0.14
[ INFO] 2016-08-27 21:47:17 : cov75                          2604
[ INFO] 2016-08-27 21:47:17 : p cov75                        0.08
[ INFO] 2016-08-27 21:47:17 : cov85                          2067
[ INFO] 2016-08-27 21:47:17 : p cov85                        0.07
[ INFO] 2016-08-27 21:47:17 : cov95                          1557
[ INFO] 2016-08-27 21:47:17 : p cov95                        0.05
[ INFO] 2016-08-27 21:47:17 : reference coverage             0.17

transrate --assembly /mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/reference/matz/past.fasta --threads=8 --reference Porites_astreoides_larvae.Trinity.fixed.fasta --output matz_Past_v_P_ast_salinity


[ INFO] 2016-08-27 21:48:44 : Contig metrics:
[ INFO] 2016-08-27 21:48:44 : -----------------------------------
[ INFO] 2016-08-27 21:48:44 : n seqs                        30740
[ INFO] 2016-08-27 21:48:44 : smallest                        100
[ INFO] 2016-08-27 21:48:44 : largest                        8171
[ INFO] 2016-08-27 21:48:44 : n bases                    16907062
[ INFO] 2016-08-27 21:48:44 : mean len                     539.09
[ INFO] 2016-08-27 21:48:44 : n under 200                    2164
[ INFO] 2016-08-27 21:48:44 : n over 1k                      3274
[ INFO] 2016-08-27 21:48:44 : n over 10k                        0
[ INFO] 2016-08-27 21:48:44 : n with orf                     7498
[ INFO] 2016-08-27 21:48:44 : mean orf percent              67.12
[ INFO] 2016-08-27 21:48:44 : n90                             332
[ INFO] 2016-08-27 21:48:44 : n70                             447
[ INFO] 2016-08-27 21:48:44 : n50                             686
[ INFO] 2016-08-27 21:48:44 : n30                             994
[ INFO] 2016-08-27 21:48:44 : n10                            1711
[ INFO] 2016-08-27 21:48:44 : gc                             0.42
[ INFO] 2016-08-27 21:48:44 : gc skew                       -0.02
[ INFO] 2016-08-27 21:48:44 : at skew                       -0.02
[ INFO] 2016-08-27 21:48:44 : cpg ratio                      1.71
[ INFO] 2016-08-27 21:48:44 : bases n                         924
[ INFO] 2016-08-27 21:48:44 : proportion n                    0.0
[ INFO] 2016-08-27 21:48:44 : linguistic complexity          0.12
[ INFO] 2016-08-27 21:48:44 : Contig metrics done in 4 seconds
[ INFO] 2016-08-27 21:48:44 : No reads provided, skipping read diagnostics
[ INFO] 2016-08-27 21:48:44 : Calculating comparative metrics...
[ INFO] 2016-08-27 21:49:17 : Comparative metrics:
[ INFO] 2016-08-27 21:49:17 : -----------------------------------
[ INFO] 2016-08-27 21:49:17 : CRBB hits                     15086
[ INFO] 2016-08-27 21:49:17 : n contigs with CRBB           15086
[ INFO] 2016-08-27 21:49:17 : p contigs with CRBB            0.49
[ INFO] 2016-08-27 21:49:17 : rbh per reference              0.25
[ INFO] 2016-08-27 21:49:17 : n refs with CRBB              13544
[ INFO] 2016-08-27 21:49:17 : p refs with CRBB               0.22
[ INFO] 2016-08-27 21:49:17 : cov25                          6121
[ INFO] 2016-08-27 21:49:17 : p cov25                         0.1
[ INFO] 2016-08-27 21:49:17 : cov50                          4801
[ INFO] 2016-08-27 21:49:17 : p cov50                        0.08
[ INFO] 2016-08-27 21:49:17 : cov75                          3407
[ INFO] 2016-08-27 21:49:17 : p cov75                        0.06
[ INFO] 2016-08-27 21:49:17 : cov85                          2861
[ INFO] 2016-08-27 21:49:17 : p cov85                        0.05
[ INFO] 2016-08-27 21:49:17 : cov95                          2117
[ INFO] 2016-08-27 21:49:17 : p cov95                        0.03
[ INFO] 2016-08-27 21:49:17 : reference coverage              0.1

#### salmon

salmon index --index mansour_Past --transcripts /mnt/scratch/ljcohen/hboi/P_astreoides/RNAseq/reference/Por_ast_ann_allTrans.fasta --type quasi
salmon index --index salinity_Past --transcripts /mnt/scratch/ljcohen/hboi/P_astreoides/RNAseq/Porites_astreoides_larvae.Trinity.fixed.fasta --type quasi

salmon quant -i mansour_Past --libType IU \
-1 /mnt/scratch/ljcohen/hboi/P_astreoides/2016/trim/25ppt.trim_1P.fq \
-2 /mnt/scratch/ljcohen/hboi/P_astreoides/2016/trim/25ppt.trim_2P.fq \
-o 25ppt.quant \
--dumpEq --auxDir aux


salmon quant -i mansour_Past --libType IU \
-1 /mnt/scratch/ljcohen/hboi/P_astreoides/2016/trim/30ppt.trim_1P.fq \
-2 /mnt/scratch/ljcohen/hboi/P_astreoides/2016/trim/30ppt.trim_2P.fq \
-o 30ppt.quant \
--dumpEq --auxDir aux

salmon quant -i mansour_Past --libType IU \
-1 /mnt/scratch/ljcohen/hboi/P_astreoides/2016/trim/35ppt.trim_1P.fq \
-2 /mnt/scratch/ljcohen/hboi/P_astreoides/2016/trim/35ppt.trim_2P.fq \
-o 35ppt.quant \
--dumpEq --auxDir aux


### Annotation
### Align all trimmed reads to both Mansour P_ast only and Symbiodinium only
get bam files

module load GNU/4.8.3
module load SAMTools/1.3.1
module load bwa/0.7.13.r1126

bwa index /mnt/home/ljcohen/Porites_astreoides_larvae_salinity/RNAseq/reference/trinityVsSymb_best.sig.fasta
bwa index /mnt/home/ljcohen/Porites_astreoides_larvae_salinity/RNAseq/reference/p_ast2016.fasta

# Symbiodinium reads

bwa mem /mnt/home/ljcohen/Porites_astreoides_larvae_salinity/RNAseq/reference/trinityVsSymb_best.sig.fasta \
/mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trim/25ppt.trim_1P.fq \
/mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trim/25ppt.trim_1P.fq > P_ast_salinity_25ppt_Symbiodinium.bam

bwa mem /mnt/home/ljcohen/Porites_astreoides_larvae_salinity/RNAseq/reference/trinityVsSymb_best.sig.fasta \
/mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trim/30ppt.trim_1P.fq \
/mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trim/30ppt.trim_1P.fq > P_ast_salinity_30ppt_Symbiodinium.bam

bwa mem /mnt/home/ljcohen/Porites_astreoides_larvae_salinity/RNAseq/reference/trinityVsSymb_best.sig.fasta \
/mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trim/35ppt.trim_1P.fq \
/mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trim/35ppt.trim_1P.fq > P_ast_salinity_35ppt_Symbiodinium.bam

# Porites astreoides only
bwa mem /mnt/home/ljcohen/Porites_astreoides_larvae_salinity/RNAseq/reference/p_ast2016.fasta \
/mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trim/25ppt.trim_1P.fq \
/mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trim/25ppt.trim_1P.fq > P_ast_salinity_25ppt_pastonly.bam

bwa mem /mnt/home/ljcohen/Porites_astreoides_larvae_salinity/RNAseq/reference/p_ast2016.fasta \
/mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trim/30ppt.trim_1P.fq \
/mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trim/30ppt.trim_1P.fq > P_ast_salinity_30ppt_pastonly.bam

bwa mem /mnt/home/ljcohen/Porites_astreoides_larvae_salinity/RNAseq/reference/p_ast2016.fasta \
/mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trim/35ppt.trim_1P.fq \
/mnt/research/ged/lisa/hboi/NGS/P_astreoides_salinity_exposure/2016/trim/35ppt.trim_1P.fq > P_ast_salinity_35ppt_pastonly.bam

#### annotate everything

# forward
makeblastdb -dbtype nucl -input_type fasta -in Porites_astreoides.trinity.fixed.fasta
- Tamer's 2016 file
- Tamer's all file
- Tamer's Symbiodinium file



# tblastn translated nucleotide as database using protein query
- Anthozoa

[ljcohen@dev-intel14 protein]$ grep -c "^>" uniprot-anthozoa.fasta
37756
(473 reviewed Swiss-Prot, unreviewed 37,283 TrEMBL Anthozoa )
Hexacorallia (31899)
Actiniaria (sea anemones) (26083)
Antipatharia (black corals) (549)
Ceriantharia (tube-dwelling anemones) (32)
Corallimorpharia (243)
Dactylanthus antarcticus (1)
Scleractinia (stony corals) (4603)
Zoantharia (zoanthids) (388)
Octocorallia (5769)
Alcyonacea (soft corals) (5549)
Helioporacea (19)
Pennatulacea (sea pens) (193)
Telestidae (8)
Relicanthus daphneae (1)


- Dinophyceae

Cryptophyta (7)
Cryptomonadaceae (2)
Cryptophyta sp. CR-MAL11 (1)
Pyrenomonadales (4)
Dinophyceae (5119)
Blastodinium crassum (11)
Dinophysiaceae (150)
environmental samples (24)
Gonyaulacales (660)
Gymnodiniales (1735)
Noctiluca scintillans (Sea sparkle) (Dinoflagellate) (91)
Oxyrrhis (100)
Peridiniales (1074)
Prorocentraceae (375)
Pyrocystis (36)
Spiniferodinium (5)
Suessiales (767)
Suessiaceae (25)
Symbiodinium (742)
Symbiodinium aenigmaticum (1)
Symbiodinium antillogorgium (1)
Symbiodinium cf. thermophilum (9)
Symbiodinium endomadracis (2)
Symbiodinium goreaui (12)
Symbiodinium kawagutii (Dinoflagellate) (22)
Symbiodinium microadriaticum (Dinoflagellate) (Zooxanthella microadriatica) (33)
Symbiodinium minutum (4)
Symbiodinium muscatinei (Dinoflagellate) (9)
Symbiodinium pseudominutum (1)
Symbiodinium psygmophilum (2)
Symbiodinium pulchrorum (Dinoflagellate) (4)
Symbiodinium sp. (Dinoflagellate) (6)
Symbiodinium sp. clades (410)
Symbiodinium thermophilum (10)
Symbiodinium trenchii (1)
Symbiodinium voratum (1)
unclassified Symbiodinium (214)
Syndiniales (59)
unclassified Dinophyceae (30)
Woloszynskia (2)
Florideophyceae sp. ex Dinophysis acuminata (4)

tblastn \
-query /mnt/home/ljcohen/protein/uniprot-anthozoa.fasta \
-db Porites_astreoides.trinity.fixed.fasta \
-out P_ast_salinity_vs_anthozoa.tab \
-evalue 1e-5 \
-outfmt 7

transdecoder to translate

# reverse
blastp protein database with translated nuleotides

#### Now want to find the extra stuff. You've assembled everything in the Mansour assembly plus extra. What is that extra?

write a script to compare contigs not in blast using the contigs.csv

sort by length
filter by has_crb = FALSE

biological filtering with protein verification:

### need transrate score for trimmed reads against your transcriptome
### need transrate score for trimmed reads against Tamer's transcriptome


transrate --assembly=Porites_astreoides.trinity.fixed.fasta --threads=16 --left=/mnt/scratch/ljcohen/hboi/P_astreoides/2016/trim/all_1P.fq --right=/mnt/scratch/ljcohen/hboi/P_astreoides/2016/trim/all_2P.fq --output trim.P_ast.score

transrate --assembly=reference/Por_ast_ann_allTrans.fasta --threads=16 --left=/mnt/scratch/ljcohen/hboi/P_astreoides/2016/trim/all_1P.fq --right=/mnt/scratch/ljcohen/hboi/P_astreoides/2016/trim/all_2P.fq --output trim.mansour_P_ast.score

