---
title: "HBOI, Edge lab, González Angel thesis, Porites astreoides larvae salinity challenge"
author: "Lisa Cohen"
output: html_document
---

# Porites astreoites microarray

Script used to generate this document:

https://github.com/ljcohen/Porites_astreoites_larvae_salinity/blob/master/microarray/Past_salinity_2016-07-01_featureAVG_probeset_analysis.Rmd

QC report:

https://ljcohen.github.io/Porites_astreoites_larvae_salinity/microarray/QC/salinity_ArrayQualityMetrics_log_collapsed_by_transcriptName_quantilenorm_filtered/index.html

Import R/Bioconductor libraries, set path to directory where raw data files are located, and overLapper venn diagram script:

```{r,echo=FALSE, message=FALSE, warning=FALSE}
library(limma)
# this creates QC reports
# library("arrayQualityMetrics")
library("Biobase")
library("RColorBrewer")
library("gplots")
library(ggplot2)
library(lattice)
setwd("~/Documents/HBOI/Past_salinity/salinity_microarray/")
source('~/Documents/scripts/overLapper.R')
```

F532 Median - B532 data were manually extracted from GenePix .txt files, all samples compiled together manually into .csv. (In the future, data extraction can be automated. We no longer have the original .txt files.) Dimensions should be 15744 x 16:
```{r,echo=FALSE, message=FALSE, warning=FALSE}
salinity_data<-read.csv("Past_salinity_microarray_data.csv")
salinity_dataframe<-data.frame(salinity_data)
salinity_dataframe_sort<-salinity_dataframe[order(salinity_dataframe$RefNum),]
#head(salinity_dataframe_sort)
rownames(salinity_dataframe_sort)<-salinity_dataframe_sort$RefNumber
#head(salinity_dataframe_sort)
#colnames(salinity_dataframe_sort)
#head(salinity_dataframe_sort)
dim(salinity_dataframe_sort)
```

Get rid of data that were flagged "BAD" = -100. Dimensions should be 15732 x 16:
```{r,echo=FALSE, message=FALSE, warning=FALSE}
salinity.data.filtered.flags <- salinity_dataframe_sort[salinity_dataframe_sort$Flags!=-100,]
#head(salinity.data.filtered.flags)
dim(salinity.data.filtered.flags)
```

Import and merge annotations file.
```{r,echo=FALSE, message=FALSE, warning=FALSE}
annotation<-read.csv("FeatAnnotFile_SEE_10-18-12.csv")
annotation.filtered<-annotation[,][match(rownames(salinity.data.filtered.flags),annotation$RefNumber),]
dim(annotation.filtered)
```

```{r,echo=FALSE, message=FALSE, warning=FALSE}
#dim(salinity.data.filtered.flags)
colnames(salinity.data.filtered.flags)
tmp_data<-salinity.data.filtered.flags[,c(8:16)]
#dim(tmp_data)
#head(salinity.data.filtered.flags)
#plotMD(tmp_data)
```

Remove controls, display number of annotations that are not controls. Should be 12,354:
```{r,echo=FALSE, message=FALSE, warning=FALSE}
annot.keep<-annotation[annotation$ControlType=="FALSE",]
#colnames(annot.keep)
dim(annot.keep)
```

Displays number of unique gene names, should be 2206:
```{r,echo=FALSE, message=FALSE, warning=FALSE}
length(unique(annot.keep$Acc))
```

Remove data rows corresponding to control probes. It is a good idea to look at control probe intensities. (Need to figure out how?) Collapse to both probe replicate and probeset levels: 1. collapse all probe reps, 2) probe rep = ID col, 3) group by ID (should be 3 reps of each ID, maybe less filtered). This is the number of unique ID, should be 4119. Take the mean of all rep intensities:
```{r,echo=FALSE, message=FALSE, warning=FALSE}
salinity.data.filtered<-salinity.data.filtered.flags[,][match(annot.keep$RefNumber,rownames(salinity.data.filtered.flags)),]
u<-unique(salinity.data.filtered$ID)
#length(u)
salinity.data.probe.rep<-aggregate(salinity.data.filtered[,c(8:16)], by=salinity.data.filtered['ID'], mean)
#class(salinity.data.probe.rep)
dim(salinity.data.probe.rep)
```

Example ID syntax: CUST_3_PI426266566
Example Name syntax: 75858820
There are 9 replicates of each name, probes designed at 3 positions on each transcript, replicated 3 times. This is the number of unique Names:

```{r,echo=FALSE, message=FALSE, warning=FALSE}
# 2. Take ID column in annotation, split by "_"
#salinity.data.probeset.ID <- strsplit(as.character(salinity.data.filtered$ID),'_',fixed=TRUE)
# Get unique 3rd position in ID
#v<-unique(sapply(strsplit(as.character(salinity.data.filtered$ID),'_',fixed=TRUE),"[",3))
#length(v)
x<-unique(salinity.data.filtered$Name)
length(x)
salinity.probeset.group<-aggregate(salinity.data.filtered[,c(8:16)], by=list(sapply(strsplit(as.character(salinity.data.filtered$ID),'_',fixed=TRUE),"[",3)), mean)
#dim(salinity.probeset.group)
salinity.probeset.name<-aggregate(salinity.data.filtered[,c(8:16)], by=salinity.data.filtered['Name'], mean)
dim(salinity.probeset.name)
```

Analyze data collapsed by name = salinity.probeset.name. This is "probeset" = same trancript/gene. Mean of intensities for all matching third position:
 
```{r,echo=FALSE, message=FALSE, warning=FALSE}
# probeset is 3rd position
# group by third position
# To-Do: look at variation between individual probes for each Name 
head(salinity.probeset.name)
dim(salinity.probeset.name)
# put Name as rownames
rownames(salinity.probeset.name)<-gsub("[[:space:]]", "",salinity.probeset.name$Name)
#colnames(salinity.probeset.name)
# Remove Name so data frame is only data:
salinity.data<-salinity.probeset.name[,c(2:10)]
salinity.data<-salinity.data[,-9]
#colnames(salinity.data)
#head(salinity.data)
```

Set all negative values to 0, transform all data + 1, # log base2 transform all data:
```{r,echo=FALSE, message=FALSE, warning=FALSE}
salinity.data[salinity.data<0]=0
#head(salinity.data)
salinity.data<-salinity.data+1
#head(salinity.data)
salinity.data.log<-log2(salinity.data)
#head(salinity.data.log)
salinity.data.matrix<-as.matrix(salinity.data.log)
salinity.data.matrix<-na.omit(salinity.data.matrix)
#head(salinity.data.matrix)
salinity.data.matrix<-salinity.data.matrix[order(rownames(salinity.data.matrix)),]
#head(annot.keep)
annot.keep<-subset(annot.keep,annot.keep$Name %in% rownames(salinity.data.matrix))
annot.keep<-annot.keep[!duplicated(annot.keep$Name),]
#dim(annot.keep)
head(annot.keep)
rownames(annot.keep)<-gsub("[[:space:]]", "", annot.keep$Name)
annot.keep<-annot.keep[order(rownames(annot.keep)),]
#head(annot.keep)
# example from limma manual:
# RG <- read.maimages(files,source="genepix",
#+ columns=list(G="F532 Median",Rb="B635",Gb="B532"))
# what format is RG in? I don't have that, so how can I proceed to the next step?
# RG <- backgroundCorrect(RG, method="normexp", offset=50)
# MA <- normalizeWithinArrays(RG)
```

Normalize between arrays:
```{r,echo=FALSE, message=FALSE, warning=FALSE}
# Need to figure out how to normalize within each array without background values from GenePix .txt files
#normDat_within<-normalizeWithinArrays(salinity.data.matrix,method="loess")
# quantile looks slightly better in arrayQualityMetrics
normDat<-normalizeBetweenArrays(salinity.data.matrix,method="quantile")
# Loess not so bad either, but want to avoid overfitting
#normDat<-normalizeBetweenArrays(salinity.data.matrix,method="cyclicloess")
#head(normDat)
salinity.norm<-normDat
#salinity.norm<-normalize.quantiles(salinity.data.matrix)
#salinity.norm<-normalizeCyclicLoess(salinity.data.matrix)
#head(salinity.norm)
#salinity.norm <- backgroundCorrect(salinity.data.matrix, method="normexp", normexp.method="saddle",offset=5)
#dim(salinity.norm)
```

The dimensions of the filtered data frame after normalization:
```{r,echo=FALSE, message=FALSE, warning=FALSE}
#coln<-colnames(salinity.data.matrix)
#rown<-rownames(salinity.data.matrix)
#colnames(salinity.norm)<-coln
#rownames(salinity.norm)<-rown
#salinity.norm<-na.omit(salinity.norm)
#class(salinity.norm)
# filter after normalization
#dim(salinity.norm)
# average signal probes
#dim(salinity.norm[rowMeans(salinity.norm)>1,])
salinity.norm.filtered<-salinity.norm[rowMeans(salinity.norm)>1,]
norm.data.filtered<-salinity.norm.filtered
#head(norm.data.filtered)
dim(norm.data.filtered)
#colnames(norm.data.filtered)
```

PCA
```{r,echo=FALSE, message=FALSE, warning=FALSE}
# PCA
x<-norm.data.filtered
colnames(x)
colnames(x)<-c("sal25ppt_A","sal25ppt_B","sal25ppt_C","sal30ppt_A","sal30ppt_B","sal30ppt_C","sal35ppt_A","sal35ppt_B")
colnames(x)
pca = prcomp(t(x))
names = colnames(x)
#fac = factor(sapply(names,function(x){strsplit(x,'Rep')[[1]][1]}))
fac= factor(c("sal25ppt","sal25ppt","sal25ppt","sal30ppt","sal30ppt","sal30ppt","sal35ppt","sal35ppt"))
colours = c("red","blue","green")
xyplot(
  PC2 ~ PC1, groups=fac, data=as.data.frame(pca$x), pch=16, cex=1.5,
  panel=function(x, y, ...) {
    panel.xyplot(x, y, ...);
    ltext(x=x, y=y, labels=names, pos=1, offset=0.8, cex=1)
  },
  aspect = "fill", col=colours
  #main = draw.key(key = list(rect = list(col = list(col=colours), text = list(levels(fac)), rep = FALSE)))
)
```


```{r,echo=FALSE, message=FALSE, warning=FALSE}

annot.keep<-subset(annot.keep,rownames(annot.keep) %in% rownames(norm.data.filtered))
# unfiltered:
#annot.keep<-subset(annot.keep,rownames(annot.keep) %in% rownames(salinity.norm))
an<-new("AnnotatedDataFrame",data=annot.keep)
experimentData<-new("MIAME",name="Sara Edge, Lisa Cohen, Ana María González Angel",
                    lab="Marine Genomics, HBOI",
                    contact="lisa.johnson.cohen@gmail.com",
                    title="Porites astreoides larvae actuve salinity exposure: 25 ppt, 30 ppt, 35 ppt",
                    abstract="ExpressionSet for P. astreoides Microarray Data",
                    other=list(notes="Extracted from raw data with GenePixPro software"))
#sample_type<-c("sal25ppt","sal25ppt","sal25ppt","sal30ppt","sal30ppt","sal30ppt","sal35ppt","sal35ppt","sal35ppt")
# remove: "c253295110054_ar2_jul1112_C_C"
sample_type<-c("sal25ppt","sal25ppt","sal25ppt","sal30ppt","sal30ppt","sal30ppt","sal35ppt","sal35ppt")

sample_name<-c("c253295110048_ar2_jun1912_A_SL","c253295110048_ar1_jun1912_B_SL","c253295110048_ar4_jun1912_C_SL",
               "c253295110048_ar3_jun1912_A_L","c253295110048_ar6_jun1912_B_L","c253295110048_ar5_jun1912_C_L",
               "c253295110048_ar8_jun1912_A_C","c253295110048_ar7_jun1912_B_C")
sample_info<-cbind(sample_type,sample_name)
sample_info<-as.data.frame(sample_info)
rownames(sample_info)<-sample_name
#sample_info
pd<-new("AnnotatedDataFrame",data=sample_info)

# unfiltered, unnormalized
#salinity_ExpressionSet_unfiltered<-new("ExpressionSet",exprs=salinity.norm,phenoData=pd,experimentData=experimentData,featureData=an)
# filtered, normalized
salinity_ExpressionSet_filtered<-new("ExpressionSet",exprs=norm.data.filtered,phenoData=pd,experimentData=experimentData,featureData=an)
#arrayQualityMetrics(expressionset = salinity_ExpressionSet_filtered,outdir = "salinity_ArrayQualityMetrics_log_collapsed_by_transcriptName_loessnorm_filtered_one35pptoutlierremoved",force = TRUE,intgroup = c("sample_type"),do.logtransform=FALSE)
###
design <- model.matrix(~ 0+factor(c(1,1,1,2,2,2,3,3)))
colnames(design) <- c("sal35ppt", "sal30ppt", "sal25ppt")
design
fit <- lmFit(salinity_ExpressionSet_filtered,design)
contrast.matrix <- makeContrasts(sal25ppt-sal30ppt, sal25ppt-sal35ppt, sal30ppt-sal35ppt, levels= design)
#contrast.matrix
fit2 <- contrasts.fit(fit, contrast.matrix)
fit2 <- eBayes(fit2)
fit2
tt_25ppt_v_30ppt<-topTable(fit2,coef="sal25ppt - sal30ppt", n=1000,adjust="BH")
tt_25ppt_v_35ppt<-topTable(fit2,coef="sal25ppt - sal35ppt",n=1000,adjust="BH")
tt_30ppt_v_35ppt<-topTable(fit2,coef="sal30ppt - sal35ppt",n=1000,adjust="BH")
#colnames(tt_25ppt_v_30ppt)
norm.data.reps<-cbind(norm.data.filtered,rownames(norm.data.filtered))
#dim(norm.data.reps)
colnames(norm.data.reps)<-c("c253295110048_ar2_jun1912_A_SL","c253295110048_ar1_jun1912_B_SL","c253295110048_ar4_jun1912_C_SL",                  "c253295110048_ar3_jun1912_A_L","c253295110048_ar6_jun1912_B_L","c253295110048_ar5_jun1912_C_L", "c253295110048_ar8_jun1912_A_C","c253295110048_ar7_jun1912_B_C","Name")
#colnames(norm.data.reps)
#head(norm.data.reps)
```

```{r,echo=FALSE, message=FALSE, warning=FALSE}
tt_25ppt_v_30ppt_reps<-merge(norm.data.reps,tt_25ppt_v_30ppt,by="Name")
tt_25ppt_v_35ppt_reps<-merge(norm.data.reps,tt_25ppt_v_35ppt,by="Name")
tt_30ppt_v_35ppt_reps<-merge(norm.data.reps,tt_30ppt_v_35ppt,by="Name")
#dim(tt_25ppt_v_30ppt_reps)
#dim(tt_25ppt_v_35ppt_reps)
#dim(tt_30ppt_v_35ppt_reps)
#write.csv(tt_25ppt_v_30ppt_reps,"P_ast_microarray_25pptvs30ppt_reps_byName_quantile.csv")
#write.csv(tt_25ppt_v_35ppt_reps,"P_ast_microarray_25pptvs35ppt_reps_byName_quantile.csv")
# filter by padj<0.25
tt_25ppt_v_30ppt_sig<-subset(tt_25ppt_v_30ppt_reps,tt_25ppt_v_30ppt_reps$adj.P.Val<0.1)
tt_25ppt_v_35ppt_sig<-subset(tt_25ppt_v_35ppt_reps,tt_25ppt_v_35ppt_reps$adj.P.Val<0.1)
tt_30ppt_v_35ppt_sig<-subset(tt_30ppt_v_35ppt_reps,tt_30ppt_v_35ppt_reps$padj.P.Val<0.1)
tt_25ppt_v_30ppt_sig<-tt_25ppt_v_30ppt_sig[order(-tt_25ppt_v_30ppt_sig$adj.P.Val),]
tt_25ppt_v_35ppt_sig<-tt_25ppt_v_35ppt_sig[order(-tt_25ppt_v_35ppt_sig$adj.P.Val),]
tt_30ppt_v_35ppt_sig<-tt_30ppt_v_35ppt_sig[order(-tt_30ppt_v_35ppt_sig$adj.P.Val),]
#dim(tt_25ppt_v_30ppt_sig)
#dim(tt_25ppt_v_35ppt_sig)
#dim(tt_30ppt_v_35ppt_sig)
#colnames(tt_25ppt_v_35ppt_sig)
#dim(subset(tt_25ppt_v_35ppt_sig,logFC< -1 | logFC>1))
logFC_1<-subset(tt_25ppt_v_35ppt_sig,logFC< -1 | logFC>1)
#head(logFC_1)
#dim(logFC_1)
#write.csv(logFC_1,"P_ast_salinity_67sig_logFC1.csv")
#write.csv(logFC_1,"P_ast_salinity_54sig_logFC1.csv")
write.csv(logFC_1,"P_ast_salinity_sig_logFC1.csv")
setlist <- list(sal25ppt_v_sal30ppt_padj0.25=as.vector(tt_25ppt_v_30ppt_sig$Name),sal25ppt_v_sal35ppt_padj0.25=as.vector(tt_25ppt_v_35ppt_sig$Name),sal25ppt_v_sal30ppt_padj0.25=as.vector(tt_25ppt_v_30ppt_sig$Name))
OLlist <- overLapper(setlist=setlist, sep="", type="vennsets")
counts <- sapply(OLlist$Venn_List, length)
vennPlot(counts=counts)
sal25ppt_35ppt_overlap<-intersect(tt_25ppt_v_30ppt_sig$Name,tt_25ppt_v_35ppt_sig$Name)
#length(sal25ppt_35ppt_overlap)
#####
```

Make heatmap of 25ppt vs. 35 ppt, log2FC > 1 and log2FC < -1
```{r,echo=FALSE, message=FALSE, warning=FALSE}
#head(logFC_1)
#dim(logFC_1)
d <- as.matrix(logFC_1[,c(2:9)])
class(d)<-"numeric"
rownames(d) <- logFC_1[,18]
#head(d)
colnames(d)<-c("25ppt_A","25ppt_B","25ppt_C","30ppt_A","30ppt_B","30ppt_C","35ppt_A","35ppt_B")
hr <- hclust(as.dist(1-cor(t(d), method="pearson")), method="complete")
mycl <- cutree(hr, h=max(hr$height/1.5))
clusterCols <- rainbow(length(unique(mycl)))
myClusterSideBar <- clusterCols[mycl]
myheatcol <- greenred(75)
#png("Salinity_heatmap_1.png", width = 7*300,height = 7*300,res = 1200,pointsize = 2)
heatmap.2(d, main="25ppt vs. 35 ppt (padj<0.05,log2FC+-1)", 
          Rowv=as.dendrogram(hr),
          cexRow=0.5,cexCol=1,srtCol= 90,
          adjCol = c(NA,0),offsetCol=2, 
          Colv=NA, dendrogram="row", 
          scale="row", col=myheatcol, 
          density.info="none", 
          trace="none", RowSideColors= myClusterSideBar)
#dev.off()
```

Versions:
```{r,echo=FALSE, message=FALSE, warning=FALSE}
sessionInfo()
```

References:

Ritchie, ME, Phipson, B, Wu, D, Hu, Y, Law, CW, Shi, W, and Smyth, GK (2015).
limma powers differential expression analyses for RNA-sequencing and microarray studies.
Nucleic Acids Research 43(7), e47.


Smyth, G. K. (2004). Linear models and empirical Bayes methods for assessing differential
expression in microarray experiments. Statistical Applications in Genetics and Molecular
Biology, Vol. 3, No. 1, Article 3.
http://www.statsci.org/smyth/pubs/ebayes.pdf

limma Manual:
https://www.bioconductor.org/packages/devel/bioc/vignettes/limma/inst/doc/usersguide.pdf

limma Bioconductor:
https://bioconductor.org/packages/release/bioc/html/limma.html
